src/interpreter/ByteCode.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+
+#include "interpreter/ByteCode.h"
+#include "runtime/ObjectType.h"
+
+namespace Walrus {
+
+// Return static size of opcodes
+size_t ByteCode::getBaseSize()
+{
+    OpcodeKind opcode = this->opcode();
+
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0);
+
+    switch (opcode) {
+    case BrTableOpcode: {
+        return g_byteCodeInfo[BrTableOpcode].m_size + sizeof(int32_t) * static_cast<BrTable*>(this)->tableSize();
+    }
+    case CallIndirectOpcode: {
+        CallIndirect* callIndirect = static_cast<CallIndirect*>(this);
+        size_t paramSize = sizeof(ByteCodeStackOffset) * static_cast<CallIndirect*>(this)->functionType()->param().size();
+        size_t resultSize = sizeof(ByteCodeStackOffset) * static_cast<CallIndirect*>(this)->functionType()->result().size();
+
+        return g_byteCodeInfo[CallIndirectOpcode].m_size + paramSize + resultSize;
+    }
+    default: {
+        return g_byteCodeInfo[opcode].m_size;
+    }
+    }
+}
+
+// Returns real size of generated opcodes
+size_t ByteCode::getSize(FunctionType* function)
+{
+    OpcodeKind opcode = this->opcode();
+
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0 && function != nullptr);
+
+    switch (opcode) {
+    case EndOpcode: {
+        return g_byteCodeInfo[EndOpcode].m_size + sizeof(ByteCodeStackOffset) * function->result().size();
+    }
+    case ThrowOpcode: {
+        return g_byteCodeInfo[ThrowOpcode].m_size + sizeof(ByteCodeStackOffset) * function->result().size();
+    }
+    case CallOpcode: {
+        return g_byteCodeInfo[CallOpcode].m_size + sizeof(ByteCodeStackOffset) * function->param().size() + sizeof(ByteCodeStackOffset) * function->result().size();
```

Comment:
I think this depends on the called function, not the actual function.
BrTableOpcode, CallIndirectOpcode, etc., should be here as well.

Reply:
Please correct me if my thought process is wrong, but I believe it would be the function's user's responsibility to pass the right function to `getSize`, in this case the target of `Call`. 

Otherwise `getSize` would also need to get the whole function list, because `Call` only contains the index of the function it calls from the list.

Reply:
Yes, getSize() needs a way to access the data of all function types. The user of the function doesn't know the opcode is a call opcode, and if you would need to provide this info every time manually, that would be too complex.